ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,152 @@
+package ch.epfl.ts.evaluation
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.{ActorRef, Cancellable}
+import ch.epfl.ts.component.{ComponentRegistration, Component, ComponentRef}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Evaluates the performance of traders by total returns, volatility, draw down and sharpe ratio
+  *
+  * To use this class, redirect all previous connections into and out of the trader to
+  * instances of this class.
+  *
+  * Note that a component providing Quote data must be connected to this component in order for
+  * it to maintain a price table and calculate profit/loss correctly.
+  *
+  * @param trader the reference to the trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the initial seed money
+  * @param period the time period to compute performance
+  */
+class Evaluator(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: FiniteDuration) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
+  private val returnsList = MList[Double]()
+  private val priceTable = MMap[(Currency, Currency), Double]()
+
+  private var lastValue = initial
+  private var maxProfit = 0.0
+  private var maxLoss = 0.0
+
+  //TODO find appropriate value for risk free rate
+  val riskFreeRate = 0.03
+  def totalReturns: Double = value() / initial
+  def volatility: Double = computeVolatility
+  def drawdown: Double = maxLoss / initial
+  def sharpeRatio: Double = (totalReturns - riskFreeRate) / volatility
+
+  /**
+   * Redirects out-going connections to the trader
```

Comment:
Does this mean no one could receive messages sent by the Evaluator?
In the end we will need this component to send a final report to use it in our automatic parameter optimization (+ possibly visualize it).